docs/guide/data_types/maps.rst

```diff
@@ -37,18 +37,142 @@ may vary. SunPy can also create maps from the jpg2000 files from
 
 Creating Custom Maps
 --------------------
-It is also possible to create maps using custom data (e.g. from a simulation).
-To do this you need to provide `sunpy.map.Map <sunpy.map.map_factory.MapFactory>`
-with both the data array as well as some basic meta information. If no header is
-given then some default values are assumed. Here is a simple example::
+It is also possible to create maps using custom data (e.g. from a simulation or an observation
+from a data source that is not explicitly supported in SunPy.) To do this you need to provide
+`sunpy.map.Map <sunpy.map.map_factory.MapFactory>` with both the data array as well as appropriate
+meta information. The meta information is important as it informs the `sunpy.map.Map <sunpy.map.map_factory.MapFactory>`
+of the correct coordinate information associated with the data array. The meta information should be provided to 
+`sunpy.map.Map <sunpy.map.map_factory.MapFactory>` in the form of a header as a `dict` or `MetaDict <sunpy.util.MetaDict>`. 
+
+The keys that are required for the header information follows the `FITS standard <https://fits.gsfc.nasa.gov/fits_dictionary.html>`_. SunPy now provides a map header helper
+function to assist the user in creating a header that contains the correct meta information
+to generate a `sunpy.map.Map <sunpy.map.map_factory.MapFactory>`.
+
+The helper functionality includes a `meta_keywords <sunpy.map.header_helper.meta_keywords>` function
+that will return a `dict` of all the current meta keywords and their descriptions currently used by
+`sunpy.map.Map <sunpy.map.map_factory.MapFactory>` to make a map::
+
+    >>> from sunpy.map import header_helper
+
+    >>> header_helper.meta_keywords()
+    {'cunit1': 'Units of the coordinate increments along naxis1 e.g. arcsec **required',
+     'cunit2': 'Units of the coordinate increments along naxis2 e.g. arcsec **required', 
+     'crval1': 'Coordinate value at reference point on naxis1 **required'
+     ...
+
+The `header_helper <sunpy.map.header_helper>` functionality also includes a utility function
+`make_fitswcs_header <sunpy.map.header_helper.make_fitswcs_header>` that will return a header with the
+appropiate FITS keywords once the map data array and an `astropy.coordinates.SkyCoord` or `sunpy.coordinates.frames`
+is passed. The `astropy.coordinates.SkyCoord` or `sunpy.coordinates.frames` is defined by the
+user, and contains information on the reference coordinate, the required WCS meta and the observer location. The function returns a `sunpy.utils.MetaDict <sunpy.utils.MetaDict>`. The `astropy.coordinates.SkyCoord` or `sunpy.coordinates.frames` must contain an observation time. 
```

Comment:
What do you mean by "WCS meta" here?

Reply:
WCS information? :P

Reply:
I think that this could be reworded (as it is just talking about the SkyCoord) to say something like "reference frame, reference coordinate and observer location"?